app/models/llms/basic_workflow/prompting.py

```diff
@@ -0,0 +1,292 @@
+                f"Configuration key 'prompts_dir' must be a string, "
+                f"got {type(prompts_dir_path).__name__}"
+            )
+        
+        if not prompts_dir_path.strip():
+            raise ValueError(
+                "Configuration key 'prompts_dir' cannot be empty or whitespace-only"
+            )
+        
+        return self._validate_prompts_directory(prompts_dir_path)
+
+    def _validate_prompts_directory(self, prompts_dir_path: str):
+        try:
+            from pathlib import Path
+            prompts_dir = Path(prompts_dir_path).resolve()
+            
+            if not prompts_dir.exists():
+                raise FileNotFoundError(
+                    f"Prompts directory does not exist: {prompts_dir}\n"
+                    f"Please ensure the directory exists or update the configuration."
+                )
+            
+            if not prompts_dir.is_dir():
+                raise NotADirectoryError(
+                    f"Prompts path is not a directory: {prompts_dir}\n"
+                    f"Please ensure the path points to a directory, not a file."
+                )
+            
+            return prompts_dir
+                
+        except Exception as e:
+            raise RuntimeError(
+                f"Failed to validate prompts directory {prompts_dir_path}: {e}"
+            )
+
+    def load_prompt(self, name: str) -> str:
+        """
+        Load a prompt template from a text file.
+        
+        This method reads a prompt template file from the prompts directory.
+        The file should be named '{name}.txt' and contain the template text
+        with placeholders for dynamic values.
+        
+        Args:
+            name: The name of the prompt template file (without .txt extension)
+            
+        Returns:
+            str: The contents of the prompt template file
+            
+        Raises:
+            ValueError: If the name contains invalid characters or path traversal attempts
+            FileNotFoundError: If the prompt template file doesn't exist
+            UnicodeDecodeError: If the file contains invalid UTF-8 characters
+        """
+        sanitized_name = self._sanitize_filename(name)
+        
+        template_path = self.prompts_dir / f"{sanitized_name}.txt"
+        try:
+            resolved_path = template_path.resolve()
+            if not str(resolved_path).startswith(str(self.prompts_dir.resolve())):
+                raise ValueError(
+                    f"Path traversal attempt detected. Requested path would resolve outside "
+                    f"the prompts directory: {resolved_path}"
+                )
+        except (RuntimeError, OSError) as e:
+            raise ValueError(f"Invalid path: {e}")
+        
+        if not template_path.exists():
+            raise FileNotFoundError(f"Prompt template '{name}' not found at {template_path}")
+        
+        return template_path.read_text(encoding="utf-8")
```

Comment:
This ties prompt loading directly to file system structure, making it hard to test or extend. Abstract prompt loading to accept a loader function or interface that can be swapped for testing.